src/ch07/c1_breed_rats.py

```diff
@@ -0,0 +1,463 @@
+"""Efficiently breed rats to an average weight of 50000 grams.
+
+Use genetic algorithm on a mixed population of male and female rats.
+
+"""
+import time
+import random
+import statistics
+
+
+class BreedRats:
+    """Efficiently breed rats to an average weight of **target_wt**.
+
+    Use genetic algorithm on a mixed population of male and female rats.
+
+    Weights and number of each gender vary and can be set by modifying the
+    following:
+
+    Args:
+        num_males (int): Number of male rats in population.
+            Default is ``4``.
+        num_females (int): Number of female rats in population.
+            Default is ``16``.
+        target_wt (int): Target weight in grams. Default is ``50000``.
+        gen_limit (int): Generational cutoff to stop breeding program.
+            Default is ``500``.
+
+    """
+
+    # pylint: disable=too-many-instance-attributes,too-many-public-methods
+    # Limit is for instance attributes and public methods are 7 and 20,
+    # but analysis like this requires many constants.
+    # I am opting to make them modifiable in something that is isn't a
+    # dictionary.
+    # If there is a better way, please submit an issue for discussion.
```

Comment:
`pylint` is a wise sage that tries to warn you when you're doing too much at once. Turning this module into a class helped with usability and readability, but there are a lot of variables at work here.

While I may look into dictionaries or namedtuples again, I like that each variable can be controlled independently for testing small variances.

Testing at what point each variable stops being beneficial is an interesting pastime.